thoth/prescriptions_refresh/handlers/gh_contributors.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python3
+# thoth-prescriptions-refresh
+# Copyright(C) 2021 Fridolin Pokorny
+#
+# This program is free software: you can redistribute it and / or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Check number of contributors to projects."""
+
+import logging
+import os
+import requests
+from typing import TYPE_CHECKING
+
+from .gh_link import iter_gh_info
+
+
+if TYPE_CHECKING:
+    from thoth.prescriptions_refresh.prescriptions import Prescriptions
+
+_LOGGER = logging.getLogger(__name__)
+
+_CONTRIBUTORS_COUNT = int(os.getenv("THOTH_PRESCRIPTIONS_REFRESH_GH_CONTRIBUTORS_COUNT", 5))
+_GH_LINK_PRESCRIPTION_NAME = "gh_contributors.yaml"
+_GH_LINK_PRESCRIPTION_CONTENT = """\
+units:
+  wraps:
+  - name: {prescription_name}
+    type: wrap
+    should_include:
+      adviser_pipeline: true
+    match:
+      state:
+        resolved_dependencies:
+        - name: {package_name}
+    run:
+      justification:
+      - type: WARNING
+        message: Package '{package_name}' has less than {contributors} contributors on GitHub
+        link: {gh_link}
+        package_name: {package_name}
+"""
+
+
+def gh_contributors(prescriptions: "Prescriptions") -> None:
+    """Check number of contributors to projects."""
```

Comment:
sorry, this is a question and my first try to comment in the review. You have  "Prescriptions" as a parameter type. What is the purpose?

Reply:
The Prescription class encapsulates operations on top of prescriptions available. If you are asking about the string representing the type hint - in this case it is not that needed. It will save some Python interpreter cycles as the import statement will not be executed (it is not much though, just for clarity the import is for type hinting):

https://github.com/thoth-station/prescriptions-refresh-job/pull/17/files#diff-730a8520d56e276c0667812d96ad46ff7dd1b9ba14f8dde61e5ee64a1a55bb3bR28-R29

The string representation of a type hint is useful when dealing with cyclic imports (to break the cycles) or when you have an implementation that references a type before it is actually defined - see for example https://stackoverflow.com/questions/51629105/python-type-hint-given-as-a-string for an example.